src/Extension/FluentExtension.php

```diff
@@ -593,16 +594,18 @@ public function augmentSQL(SQLSelect $query, DataQuery $dataQuery = null)
      * loading a page in a different locale and pressing "save" won't actually make the record available in
      * this locale.
      */
-    public function onBeforeWrite()
+    public function onBeforeWrite(): void
     {
         $owner = $this->owner;
-
-        /** @var string $currentLocale */
         $currentLocale = FluentState::singleton()->getLocale();
+
         if (!$currentLocale) {
             return;
         }
 
+        // Detect class name change (used later to properly localised newly available relation data)
+        $owner->classNameChangeDetected = $owner->isInDB() && $owner->isChanged('ClassName');
```

Comment:
you can detect isChanged in onAfterWrite. I prefer to do this there than adding an unnecessary temp variable.

You can also use isChanged('ID') to see if a record was created.